proposals/0409-exportable-named-default.rst

```diff
@@ -101,10 +101,10 @@ When exporting a ``default Num`` declaration, the class ``Num`` has to be explic
 An ``import`` of a module always imports all the ``default`` declarations listed in the module's export list. There is
 no way to exclude any of them. This is the default option for this proposal, but there are `alternatives`_.
 
-A module can export its ``default`` declarations either by having no explicit export list (as in ``module M where
-{...}``) or by specifying them explicitly in its export list using the above syntax extension. In particular,
-re-export of a whole imported module (as in ``module M (module N) where{...}`` does *not* export any ``default``
-declarations.
+A module can export its ``default`` only by specifying them explicitly in its export list using the above syntax
+extension. In particular, module with no explicit export list (as in ``module M where {...}``) does *not* export any
+``default`` declarations, and neither does the re-export of a whole imported module (as in ``module M (module N)
```

Comment:
The re-export of a whole *imported* module could hardly export defaults, since the defaults in the current module are computed by combining the imported ones.

The important case is actually `module M( module M ) where...`! 

Just delete "imported" here or say "...and neither does the re-export of the whole module (as in `module M( module M ) where...`).

Reply:
It does seem like it could be surprising that the following could work:
```hs
module M (default C, ...) where
  ...

module N where
  import M

  ... some declarations needing the default ...
```
but changing `N` to import `M2` defined thus would fail:
```hs
module M2  (module M) where
  import M
```

At first glance, it looks like `M2` should be exporting an identical API to `M`, but that would not be the case under this proposal. A possible alternative would be:
 * If a module lists itself in the export list, that does not implicitly include defaults (for backwards compatibility and minimizing surprise).
 * If a module `M2` re-exports a different module `M` with a `default C` export, and does not explicitly export `default C` itself, it re-exports exactly the API of `M` (including `default C` as originally exported from `M`, not as aggregated in `M2`).
 * It is an error to re-export two modules that expose `default C` declarations but not include an explicit `default C` declaration in the export list.

But on balance I doubt this additional complication is worth it, compared to being explicit. Perhaps there could be a warning if a re-exported module has a default that is not itself being exported.